frontend/components/Sidebar.tsx

```diff
@@ -0,0 +1,192 @@
+import { Button } from 'antd';
+import { ReactElement, useMemo, useState } from 'react';
+import {
+  MenuOutlined,
+  UserOutlined,
+  BookOutlined,
+  QuestionCircleOutlined,
+  BarChartOutlined,
+} from '@ant-design/icons';
+import styled from 'styled-components';
+import slideStyle from './slide.module.scss';
+
+const MenuHeading = styled.h1`
+  color: #800080;
+`;
+
+const MenuElements = styled.div`
+  margin-left: 10px;
+  :hover {
+    color: #800080;
+  }
+`;
+
+const SidebarDiv = styled.div`
+  position: absolute;
+  x: 0;
+  y: 0;
+  width: 30vw;
+  height: 100vh;
+`;
+
+const MenuSection = styled.div`
+  border-bottom: 2px solid;
+`;
+
+const TwoColumns = styled.div`
+  display: flex;
+  margin-left: 10px;
+`;
+const ColumnItem = styled.div`
+  width: 50%;
+  :hover {
+    color: #800080;
+  }
+`;
+
+const Menu = styled.div`
+  width: 25vw;
+  height: 100vh;
+  display: flex;
+  flex-direction: column;
+  background: white;
+  position: absolute;
+  z-index: 0;
+`;
+
+const Sidebar = (): ReactElement => {
+  const [open, setOpen] = useState<boolean>(false);
+  const [closing, setClosing] = useState<boolean>(false);
+
+  const animation = open
+    ? slideStyle.slide
+    : closing
+    ? slideStyle.close
+    : undefined;
+  console.log(animation);
+
+  const closeMenu = () => {
+    setOpen(false);
+    setClosing(true);
+    return setTimeout(() => {
+      setClosing(false);
+    }, 500);
+  };
+
+  /**
+   * TODO: Update icon to be icon from prototype - use custom icon
+   */
+  return (
+    <SidebarDiv>
+      {open || closing ? (
+        <Menu className={animation}>
+          <Button
+            icon={<MenuOutlined />}
+            onClick={closeMenu}
+            style={{ width: '30px' }}
+            ghost
+          />
+          <MenuSection>
+            <div>
+              <MenuHeading>
+                <UserOutlined />
+                Profile
+              </MenuHeading>
+              <MenuElements>
+                <a href={'https://google.com'}>Survey Report</a>
+              </MenuElements>
+            </div>
```

Comment:
These set of tags seems to be repeated, I was wondering if we can make this into its own component to prevent repetitive code.

i.e. 
```
const Name = ({
  icon,                   // ReactElement
  iconName,         // string
  menuElements  // Array<[string, string]>
}: {...}): ReactElement => {
  return (
    <>
      <MenuHeading>
        {icon}
        {iconName}
      </MenuHeading>
      <MenuElements>
        {menuElements.map((info) => {
        const [link, name] = info;
        return (
          <>
            <a href={link}>{name}</a>
          </>
        )}}
      </MenuElements>
    </>
  );
};